components/gitpod-protocol/src/protocol.ts

```diff
@@ -70,11 +70,14 @@ export namespace User {
     }
 
     /**
-     * Tries to return the primaryEmail of the first identity this user signed up with.
+     * Returns the stored email or if it doesn't exist returns the primaryEmail of the first identity this user signed up with.
      * @param user
      * @returns A primaryEmail, or undefined if there is none.
      */
     export function getPrimaryEmail(user: User): string | undefined {
+        if (user.additionalData?.emailAddress) {
```

Comment:
While I like introducing capability of letting users change their email, and sharpening our concept for "primary email" (which is ill-conceived atm) this introduces room for fraud.
We rely upon these email-addresses to be validated (e.g. exist and are under the user's control) for student plans/bonusses.

If we merge this as is we'd have to prioritize separating these concepts.

Reply:
Maybe we could have a field `profile.email` that is chosen to be filled by the user, but only with one of the (validated) addresses from one of their auth providers.

Again, if we're fine with the temporary gap here, that can be a gap. We just need to prioritize it, then. :+1:

Reply:
This function is not used for the student eligability, so I don't think we have that problem.

Reply:
See https://github.com/gitpod-io/gitpod/blob/7c0ecdbf86a3554a59770fc61d75f0a48f0336b0/components/server/ee/src/user/eligibility-service.ts#L69

Reply:
Indeed, my bad! :+1: